src/Public/OpenApi.ps1

```diff
@@ -2933,47 +2962,56 @@ function Enable-PodeOpenApiViewer {
-
             $Data = @{
-                Title                 = $meta.Title
-                OpenApi               = $meta.OpenApi
-                OpenApiUrl            = $OpenApiUrl
-                OpenApiDefinition     = Get-PodeOpenApiDefinition | ConvertTo-Json -Depth 100
-                OpenApiYamlDefinition = Get-PodeOpenApiDefinition | ConvertTo-PodeYamlInternal -Depth 100 -NoNewLine
-                Swagger               = 'false'
-                ReDoc                 = 'false'
-                RapiDoc               = 'false'
-                StopLight             = 'false'
-                Explorer              = 'false'
-                RapiPdf               = 'false'
-            }
-            foreach ($path in ($PodeContext.Server.Routes['GET'].Keys  )) {
-                # the current route
-                $_routes = @($PodeContext.Server.Routes['GET'][$path])
-                # get the first route for base definition
-                $_route = $_routes[0]
-                # check if the route has to be published
-                if ($_route.Arguments -and $_route.Arguments.OpenApiDoc   ) {
-                    $Data[$_route.Arguments.Type] = 'true'
-                    $Data["$($_route.Arguments.Type)_path"] = $_route.Path
-                }
+                Title   = $meta.Title
+                OpenApi = $meta.OpenApi
             }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.swagger) {
+                $Data.swagger = 'true'
+                $data.swagger_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.swagger
+            } else { $Data.swagger = 'false' }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.redoc) {
+                $Data.redoc = 'true'
+                $data.redoc_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.redoc
+            } else { $Data.redoc = 'false' }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.rapidoc) {
+                $Data.rapidoc = 'true'
+                $data.rapidoc_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.rapidoc
+            } else { $Data.rapidoc = 'false' }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.stoplight) {
+                $Data.stoplight = 'true'
+                $data.stoplight_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.stoplight
+            } else { $Data.stoplight = 'false' }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.explorer) {
+                $Data.explorer = 'true'
+                $data.explorer_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.explorer
+            } else { $Data.explorer = 'false' }
+            if ($PodeContext.Server.OpenAPI.hiddenComponents.viewer.rapipdf) {
+                $Data.rapipdf = 'true'
+                $data.rapipdf_path = $PodeContext.Server.OpenAPI.hiddenComponents.viewer.rapipdf
+            } else { $Data.rapipdf = 'false' }
```

Comment:
I've just tested this locally: If you wrap the `$data.swagger` calls in the `default-doc-bookmarks.html.pode` file in single-quotes:
```js
var enableSwagger = '$($data.swagger)'; 
```

then you can shrink all these ifs down to the following 😄 
```powershell
foreach ($type in $PodeContext.Server.OpenAPI.hiddenComponents.viewer.Keys) {
    $Data[$type] = $true
    $Data["$($type)_path"] = $PodeContext.Server.OpenAPI.hiddenComponents.viewer[$type]
}
```

Reply:
if I use var enableSwagger = '$($data.swagger)'; 
enableSwagger become a string not a bool

Reply:
I made a little change to default-doc-bookmarks.html.pode
var enableSwagger = $($data.swagger.toString().ToLower());

Reply:
`$data.swagger.toString().ToLower()` will fail when one of the viewers isn't supplied, as the hashtable property will be null.

Using `var enableSwagger = '$($data.swagger)';` utilised the nulls, and the fact that empty strings in javascript return false in if-statements. Which means `enableSwagger ? "list-item" : "none";` would work when `enableSwagger` is `"True"` (returning "list-item"), and when is `""` (returning "none").

Reply:
you are right :(